### PR TITLE
Fix upgrade test

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/9010
closes #9010

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/upgrade/post/test_publish.py [hunk 1]
```diff
@@ -80,13 +80,17 @@ def test_all(self):
            repository versions to be published at same time.
         """
         # Step 1
-        for file_content in get_content(self.repo.to_dict())[FILE_CONTENT_NAME]:
+        file_content = get_content(self.repo.to_dict())[FILE_CONTENT_NAME]
+        version_hrefs = tuple(ver["pulp_href"] for ver in get_versions(self.repo.to_dict()))
+        print(f"\nContent:\n{file_content}\nVersions: \n{version_hrefs}")
+        for file_content in file_content:
```
Comment (fao89) [inline pulpcore/tests/upgrade/post/test_publish.py:86]: I put some prints for facilitating the debug

**Review by fao89 — COMMENTED** [review]
For some reason, it fails because the distribution doesn't have content:
https://github.com/pulp/pulpcore/runs/3030554001?check_suite_focus=true#step:11:1844

**Comment by fao89** [comment]
I put more prints to understand, but I couldn't see it failing again, I'll keep restart the job till I see some failure

**Comment by fao89** [comment]
the irony: the more debug helpers I add, the fewer failures I see
@dkliban should I drop the changelog entry and merge this PR so it can help us in the future?

File: pulpcore/tests/upgrade/post/test_publish.py [hunk 2]
```diff
@@ -49,9 +50,15 @@ def setUpClass(cls):
...
         cls.distributions = DistributionsFileApi(client)
+        cls.filecontent = ContentFilesApi(client)
 
     def setUp(self):
         """Create a new repository before each test."""
+        print(f"\nDistribution:\n{self.distributions.list()}")
+        print(f"\nContent:\n{self.filecontent.list()}")
+        for repo in self.repo_api.list().results:
+            for ver in get_versions(repo.to_dict()):
+                print(ver["content_summary"])
```
Comment (fao89) [inline pulpcore/tests/upgrade/post/test_publish.py:61]: I could reproduce the error:
https://github.com/fao89/pulpcore/runs/3050715281?check_suite_focus=true

the distribution and content are ok


the problem seems to be a bad sync on setUp

**Comment by fao89** [comment]
it seems the issue happens during sync:
https://github.com/fao89/pulpcore/runs/3056333980?check_suite_focus=true

File: pulpcore/tests/upgrade/pre/test_publish.py [hunk 3]
```diff
@@ -78,10 +78,15 @@ def test_all(self):
            repository versions to be published at same time.
         """
         # Step 1
-        for file_content in get_content(self.repo.to_dict())[FILE_CONTENT_NAME]:
-            modify_repo(self.cfg, self.repo.to_dict(), remove_units=[file_content])
+        repo_content = get_content(self.repo.to_dict())[FILE_CONTENT_NAME][:-1]
```
Comment (fao89) [inline pulpcore/tests/upgrade/pre/test_publish.py:81]: Assuring we always have a content (not removing all of them)

File: pulpcore/tests/upgrade/pre/test_publish.py [hunk 4]
```diff
@@ -78,10 +78,15 @@ def test_all(self):
...
+        for file_content in repo_content:
+            repository_modify_data = RepositoryAddRemoveContent(
+                remove_content_units=[file_content["pulp_href"]]
+            )
+            modify_response = self.repo_api.modify(self.repo.pulp_href, repository_modify_data)
+            monitor_task(modify_response.task)
         version_hrefs = tuple(ver["pulp_href"] for ver in get_versions(self.repo.to_dict()))
-        non_latest = choice(version_hrefs[:-1])
+        non_latest = choice(version_hrefs[1:-1])
```
Comment (fao89) [inline pulpcore/tests/upgrade/pre/test_publish.py:89]: some versions didn't have content, yielding an empty manifest, that's why the failure was intermittent

**Review by dkliban — APPROVED** [review]

**Review by dralley — APPROVED** [review]

**Pull request merged**